ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/use-fees-calculations.ts

```diff
@@ -0,0 +1,167 @@
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { EtherDenomination } from '../../../../../../../../shared/constants/common';
+import {
+  addHexes,
+  getEthConversionFromWeiHex,
+  getValueFromWeiHex,
+  multiplyHexes,
+} from '../../../../../../../../shared/modules/conversion.utils';
+import { getConversionRate } from '../../../../../../../ducks/metamask/metamask';
+import { useFiatFormatter } from '../../../../../../../hooks/useFiatFormatter';
+import { getCurrentCurrency } from '../../../../../../../selectors';
+
+export function useFeeCalculations(
+  hasLayer1GasFee: boolean,
+  layer1GasFee: string,
+  gasEstimate: string,
+  supportsEIP1559: boolean,
+  transactionMeta: TransactionMeta,
+  maxPriorityFeePerGas: string,
+  maxFeePerGas: string,
+) {
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const conversionRate = useSelector(getConversionRate);
+  const fiatFormatter = useFiatFormatter();
+
+  const getFeesFromHex = (hexFee: string) => {
+    const nativeCurrencyFee = getEthConversionFromWeiHex({
+      value: hexFee,
+      fromCurrency: EtherDenomination.GWEI,
+      numberOfDecimals: 4,
+    });
+
+    const currentCurrencyFee = fiatFormatter(
+      Number(
+        getValueFromWeiHex({
+          value: hexFee,
+          conversionRate,
+          fromCurrency: EtherDenomination.GWEI,
+          toCurrency: currentCurrency,
+          numberOfDecimals: 2,
+        }),
+      ),
+    );
+
+    return { currentCurrencyFee, nativeCurrencyFee };
+  };
+
+  const defaultFee = '';
+  const [
+    currentCurrencyEstimatedFeeFromBreakdown,
+    setCurrentCurrencyEstimatedFeeFromBreakdown,
+  ] = useState(defaultFee);
+  const [
+    nativeCurrencyEstimatedFeeFromBreakdown,
+    setNativeCurrencyEstimatedFeeFromBreakdown,
+  ] = useState(defaultFee);
+  const [currentCurrencyEstimatedFee, setCurrentCurrencyEstimatedFee] =
+    useState(defaultFee);
+  const [nativeCurrencyEstimatedFee, setNativeCurrencyEstimatedFee] =
+    useState(defaultFee);
+  const [currentCurrencyL1Fee, setCurrentCurrencyL1Fee] = useState(defaultFee);
+  const [nativeCurrencyL1Fee, setNativeCurrencyL1Fee] = useState(defaultFee);
+  const [currentCurrencyL2Fee, setCurrentCurrencyL2Fee] = useState(defaultFee);
+  const [nativeCurrencyL2Fee, setNativeCurrencyL2Fee] = useState(defaultFee);
+
+  useEffect(() => {
```

Comment:
For readability, could either of these `useEffect` / `useMemo` be moved to separate functions such as `useL1FeeCalculations`?